src/ballots/vote.cpp

```diff
@@ -0,0 +1,121 @@
+#include <ballots/vote.hpp>
+#include <dao.hpp>
+#include <common.hpp>
+#include <document_graph/edge.hpp>
+#include <member.hpp>
+#include <util.hpp>
+#include <trail.hpp>
+
+#define CONTENT_GROUP_LABEL_VOTE "vote"
+
+namespace hypha 
+{
+    Vote::Vote(hypha::dao& dao, const eosio::checksum256& hash)
+    : TypedDocument(dao, hash)
+    {
+
+    }
+    
+    Vote::Vote(
+        hypha::dao& dao, 
+        const eosio::name voter, 
+        std::string vote, 
+        Document& proposal
+    )
+    : TypedDocument(dao)
+    {
+        // Could be replaced by i.e. proposal.hasVote(vote) when proposal is no longer just a "Document"
+        // the goal is to have an easier API
+        proposal.getContentWrapper().getOrFail(BALLOT_OPTIONS, vote, "Invalid vote");
+        
+        eosio::check(
+            Edge::exists(dao.get_self(), getRoot(dao.get_self()), proposal.getHash(), common::PROPOSAL),
+            "Only allowed to vote active proposals"
+        );
+
+        std::vector<Edge> votes = dao.getGraph().getEdgesFrom(proposal.getHash(), common::VOTE);
+        for (auto votesIt = votes.begin(); votesIt != votes.end(); ++votesIt) {
```

Comment:
Mostly a curiosity question; Do you prefer this syntax over the Range-based for loop?

Reply:
No real preference, might update to the range-based one.